easybuild/easyblocks/o/openblas.py

```diff
@@ -3,18 +3,42 @@
 
 @author: Andrew Edmondson (University of Birmingham)
 """
+import os
 from easybuild.easyblocks.generic.configuremake import ConfigureMake
-from easybuild.tools.systemtools import POWER, get_cpu_architecture
+from easybuild.tools.systemtools import POWER, get_cpu_architecture, get_shared_lib_ext
 
 
 class EB_OpenBLAS(ConfigureMake):
     """Support for building/installing OpenBLAS."""
 
-    def build_step(self):
-        """Custom build procedure for OpenBLAS."""
+    def configure_step(self):
+        """ set up some options - but no configure command to run"""
+
+        if 'BINARY=' not in self.cfg['buildopts']:
+            self.cfg.update('buildopts', 'BINARY=64')
+        if 'USE_THREAD=' not in self.cfg['buildopts']:
+            self.cfg.update('buildopts', 'USE_THREAD=1')
+            self.cfg.update('installopts', 'USE_THREAD=1')
+        if 'USE_OPENMP=' not in self.cfg['buildopts']:
+            self.cfg.update('buildopts', 'USE_OPENMP=1')
+            self.cfg.update('installopts', 'USE_OPENMP=1')
+        if 'CC=' not in self.cfg['buildopts']:
+            self.cfg.update('buildopts', 'CC=%s' % os.environ['CC'])
+        if 'FC=' not in self.cfg['buildopts']:
+            self.cfg.update('buildopts', 'FC=%s' % os.environ['FC'])
```

Comment:
This is very repetitive, and can be done a lot cleaner using a `dict`:

```python
default_opts = {
    'BINARY': '64',
    'CC': os.getenv('CC'),
    'FC': os.getenv('FC'),
    'USE_OPENMP': '1',
    'USE_THREAD': '1',
}
for key in sorted(default_opts.keys()):
    for opts_key in ['buildopts', 'installopts']:
        if '%s=' % key not in self.cfg[opts_key]:
            self.cfg.update(opts_key, '%s=%s' % (key, default_opts[key]))
```

This will also add `CC=...` to `installopts`, but that shouldn't cause any problems.

Reply:
Done